variants/github_actions_ci/workflows/deploy_to_ec2.yml

```diff
@@ -0,0 +1,78 @@
+name: deploy_to_ec2
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      assume_role_arn:
+        type: string
+        required: true
+      aws_region:
+        type: string
+        required: true
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+    secrets:
+      ssh_private_key:
+        required: true
+      slack_webhook:
+        required: true
+jobs:
+  deploy_to_ec2_via_oidc:
+    name: Deploy to AWS EC2 with Capistrano, authenticated by Github OIDC
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    permissions:
+      # These permissions are needed to interact with GitHub's OIDC Token
+      # endpoint.
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Configure AWS credentials using OpenID Connect to AssumeRole
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-session-name: GitHubActionsDeployment
+          role-to-assume: ${{ inputs.assume_role_arn }}
+          aws-region: ${{ inputs.aws_region }}
+      - name: Install SSH key for production
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.ssh_private_key }}
+          known_hosts: unnecessary
+      - name: Deploy application
+        run: bundle exec cap ${{ inputs.environment }} deploy
+      - name: Report success to Slack
+        if: ${{ success() }}
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.slack_webhook }}
+          SLACK_CHANNEL: TODO
```

Comment:
I think we should make these inputs because I could see case for us wanting to send production deployment notifications to a channel with the client but not staging ones, and that way we're leveraging GitHub Actions to tell people they need to set this up (currently it'll fail when it hits here with an error about the channel not existing) which should make for a nicer experience.

Reply:
Agreed. I have updated in latest push